featuretools/primitives/standard/transform_primitive.py

```diff
@@ -623,10 +631,108 @@ class NumericLag(TransformPrimitive):
 
         return lag
+
+
+class Lag(TransformPrimitive):
+    """Shifts an array of values by a specified number of periods.
+
+    Args:
+        periods (int): The number of periods by which to shift the input.
+            Default is 1. Periods correspond to rows.
+
+        fill_value (int, float, str, bool, optional): The value to use to fill in
+            the gaps left after shifting the input. Default is None.
+
+    Examples:
+        >>> lag = Lag()
+        >>> lag([1, 2, 3, 4, 5], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
+        [nan, 1.0, 2.0, 3.0, 4.0]
+
+        You can specify the number of periods to shift the values
+
+        >>> lag_periods = Lag(periods=3)
+        >>> lag_periods(["hello", "world", "test", "foo", "bar"], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
+        [nan, nan, nan, 'hello', 'world']
+
+        You can specify the fill value to use
+
+        >>> lag_fill_value = Lag(fill_value=100)
+        >>> lag_fill_value([1, 2, 3, 4], pd.Series(pd.date_range(start="2020-01-01", periods=4, freq='D'))).tolist()
+        [100, 1, 2, 3]
+    """
+
+    name = "lag"
+    input_types = [
+        [
+            ColumnSchema(semantic_tags={"category"}),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=AgeFractional),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=AgeNullable),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=BooleanNullable),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=Double),
+            ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=EmailAddress),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=IntegerNullable),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=NaturalLanguage),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=Ordinal),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=PostalCode),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=URL),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=Unknown),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+    ]
+    return_type = None
+    uses_full_dataframe = True
+
+    def __init__(self, periods=1, fill_value=None):
```

Comment:
Not sure having a fill_value makes sense now. You couldn't specify a numeric fill value and have it work on boolean columns for example. There probably isn't much harm in leaving it with the default at `None`, but if users specified a value, I think they will have trouble unless the restrict the application of their primitive instance to only certain columns that are compatible with their specified fill value.

Reply:
Yeah I'm just never a fan of restricting functionality. I could put guard rails in by throwing TypeErrors if fill_value type doesn't match the input type. Thoughts?

Reply:
I don't really like the idea of complicating the primitive calculation with a bunch of type checking. If we are going to leave this in, maybe we just update the docstring to inform the users that errors will result if they attempt to use a fill value that is incompatible with the data types of the columns to which the primitive is applied?

Not completely sure, but I think the will end up getting a Woodwork initialization error in the event of incompatible data types. That error might not be super-helpful in troubleshooting the problem though.

Reply:
After discussing we are removing `fill_value`. Ideas to reintroduce are captured here: https://github.com/alteryx/featuretools/issues/2266